internal/host/transition_test.go

```diff
@@ -211,8 +211,17 @@ var _ = Describe("RegisterHost", func() {
 					Expect(err).ShouldNot(HaveOccurred())
 				} else {
 					Expect(err).Should(HaveOccurred())
-					serr, ok := err.(*common.ApiErrorResponse)
-					Expect(ok).Should(Equal(true))
+
+					var serr *common.ApiErrorResponse
+					switch e := err.(type) {
+					case *common.ApiErrorResponse:
+						serr = e
+					case *common.InfraErrorResponse:
+						serr = e.ApiErrorResponse
+					default:
+						Fail(fmt.Sprintf("unexpected error type: %T", err))
+					}
+
```

Comment:
Can we specify the error in the test case instead and avoid this switch?
That way we would know what to expect from each case